orchestrator/db/sorting/sorting.py

```diff
@@ -54,51 +56,69 @@ def generic_sorts_validate(
         for item in sort_by:
-            field = item.field.lower()
-            sort_key = valid_sort_dict[field]
-
-            if item.order == SortOrder.DESC.value:  # type: ignore
-                query = query.order_by(expression.desc(model.__dict__[sort_key]))
-            else:
-                query = query.order_by(expression.asc(model.__dict__[sort_key]))
+            field = item.field
+            sort_fn = valid_sort_functions_by_column[field]
+            try:
+                query = sort_fn(query, item.order)
+            except ProblemDetailException as exception:
+                handle_sort_error(
+                    exception.detail,
+                    field=field,
+                    order=item.order,
+                )
+            except ValueError as exception:
+                handle_sort_error(
+                    str(exception),
+                    field=field,
+                    order=item.order,
+                )
         return query
 
-    return _apply_sorts
+    return _apply_sorting
 
 
 def generic_sort(
-    valid_sort_dict: dict[str, str],
-    model: SqlBaseModel,
+    valid_sort_functions_by_column: ValidSortFunctionsByColumnType,
 ) -> Callable[[QueryType, list[Sort], CallableErrorHander], QueryType]:
-    valid_sort_keys = list(valid_sort_dict.keys())
-    _validate_sorts = generic_sorts_validate(valid_sort_dict)
-    _apply_sorts = generic_apply_sorts(valid_sort_dict, model)
+    valid_sort_functions_by_column_KEYS = list(valid_sort_functions_by_column.keys())
+    _validate_sorts = generic_sorts_validate(valid_sort_functions_by_column)
+    _apply_sorting = generic_apply_sorting(valid_sort_functions_by_column)
 
     def _sort(
         query: QueryType,
         sort_by: list[Sort],
         handle_sort_error: CallableErrorHander,
     ) -> QueryType:
-        if sort_by:
-            invalid_sort_items, valid_sort_items = _validate_sorts(sort_by)
-            if invalid_list := [item.dict() for item in invalid_sort_items]:
-                handle_sort_error(
-                    "Invalid sort arguments",
-                    invalid_filters=invalid_list,
-                    valid_filter_keys=valid_sort_keys,
-                )
+        invalid_sort_items, valid_sort_items = _validate_sorts(sort_by)
+        if invalid_list := [item.dict() for item in invalid_sort_items]:
+            handle_sort_error(
+                "Invalid sort arguments",
+                invalid_sorting=invalid_list,
+                valid_sort_keys=valid_sort_functions_by_column_KEYS,
+            )
 
-            query = _apply_sorts(query, valid_sort_items)
-        return query
+        return _apply_sorting(query, valid_sort_items, handle_sort_error)
 
     return _sort
+
+
+def generic_column_sort(field: Column) -> Callable[[SearchQuery, SortOrder], SearchQuery]:
+    def sort_function(query: SearchQuery, order: SortOrder) -> SearchQuery:
+        if order == SortOrder.DESC.value:  # type: ignore
```

Comment:
Well that's not true, the string comparison will probably work fine. But the `.value` shouldn't be needed right?

Reply:
I used 2 different enums, so I made the `Sort` class use `use_enum_values` which made the `.order` a string, which needs to be compared to the enum value. I made `SortOrder` into a strawberry enum and removed the `GraphqlSortOrder` enum. now we do use the enum instead of string in `Sort` class